device_backends/mmio/include/MmioAccess.h

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#pragma once
+
+#include "DeviceFile.h"
+
+#include <string>
+
+namespace ChimeraTK {
+
+  // Control (config/status) interface to access the registers of FPGA IPs
+  class MmioAccess {
+   public:
+    MmioAccess() = delete;
+    MmioAccess(const std::string& devicePath, std::size_t mapSize);
+    MmioAccess(int fd, std::size_t mapSize, bool takeFdOwnership = true);
+    ~MmioAccess();
+
+    void read(uintptr_t address, int32_t* __restrict__ buf, size_t nBytes);
+    void write(uintptr_t address, const int32_t* data, size_t nBytes);
+
+   private:
+    DeviceFile _file;
+    void* _mem{nullptr};
+
+    // Size of mmap'ed area in PCI BAR
```

Comment:
Is the comment appropriate for UIO as well? It's not PCI, right? Or is this variable only used for xdma?

Reply:
This comment is wrong. the variable is used in both, though. Fixed the comment